test/unit/transports/ssh_test.rb

```diff
@@ -55,6 +55,19 @@
     end
   end
 
+  describe 'ssh options' do
+    let(:ssh) { cls.new(conf) }
+    let(:connection) { ssh.connection }
+    it 'includes BatchMode when :non_interactive is set' do
+      conf[:non_interactive] = true
+      connection.ssh_opts.include?("BatchMode=yes").must_equal true
+    end
+    it 'excludes BatchMode when :non_interactive is not set' do
+      conf.delete :non_interactive
```

Comment:
Because `conf` is set by a `let` statement you _should_ be able to leave this line out because it is reset for every example. Could you try removing this?

Leaving it in makes these examples look like they are order dependent, which is a unit test smell IMO